gomint-api/src/main/java/io/gomint/command/Command.java

```diff
@@ -102,14 +105,17 @@ public Command( String name ) {
     }
 
     /**
-     * Execute a command for a command sender
+     * Execute a command for a command sender with the given parameters.
+     * <br><br>
+     * <b>You have to call {@linkplain CommandOutput#markFinished() output.markFinished()} at the end to indicate command
+     *  * execution finished.</b>
      *
      * @param commandSender which has executed the command
      * @param alias         which the user used to execute this command
      * @param arguments     which the player has given, optional parameters may be missing
-     * @return CommandOutput which has been generated by the execution
+     * @param output        the command output to add messages to
      */
-    public abstract CommandOutput execute(CommandSender<?> commandSender, String alias, Map<String, Object> arguments );
+    public abstract void execute(CommandSender<?> commandSender, String alias, Map<String, Object> arguments, CommandOutput output);
```

Comment:
Why is it needed to move the output to a preallocated argument?

Reply:
Constraint due to https://github.com/gomint/gomint/pull/727#discussion_r571708144 and to supply the CommandOutput consumer to the CommandOutput.